cli/command/service/opts.go

```diff
@@ -349,7 +351,31 @@ func (e *endpointOptions) ToEndpointSpec() *swarm.EndpointSpec {
 	ports, portBindings, _ := nat.ParsePortSpecs(e.ports.GetAll())
 
 	for port := range ports {
-		portConfigs = append(portConfigs, convertPortToPortConfig(port, portBindings)...)
+		rawPortConfigs := convertPortToPortConfig(port, portBindings)
```

Comment:
should we instead change `convertPortToPortConfig` to take in PublishMode instead of walking the configs again as follows ?